### PR TITLE
build buck from source and JDK 11

### PR DESCRIPTION
**Pull request by dulmandakh**
Build buck from source and use JDK 11, this will make possible to bump Android Gradle Plugin to 7.x.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+FROM ubuntu:20.04 AS buck
+
+ARG BUCK_VERSION=2021.01.12.01
+ENV ANT_OPTS="-Xmx4096m"
+RUN apt update  && apt install  -y --no-install-recommends \
+    ant \
+    git \
+    openjdk-11-jdk-headless \
+    python3 \
+    python-is-python3
```
Comment (cortinico) [inline Dockerfile:11]: I'm unsure about using python3 here as Buck requires python2 according to docs.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+# install buck by compiling it from source. We also remove the buck repo once it's built.
```
Comment (cortinico) [inline Dockerfile:12]: Please update the commen there

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+RUN git clone --depth 1 --branch v${BUCK_VERSION} https://github.com/facebook/buck.git \
+    && cd buck \
+    && ant \
+    && ./bin/buck build buck --out /tmp/buck.pex
```
Comment (cortinico) [inline Dockerfile:16]: I believe you need a `--config java.target_level=11 --config java.source_level=11` here

Comment (dulmandakh) [reply Dockerfile:16]: thanks for the comments. I'll try to build it on my local machine, hope everything works

**Comment by dulmandakh** [comment]
@cortinico successfully built React Native from source.

@gengjiawen please merge and release 5.0.

**Comment by cortinico** [comment]
Great stuff 🎉

**Pull request merged**